rcon/discord_chat.py

```diff
@@ -1,61 +1,219 @@
 import logging
 import os
-from functools import lru_cache
 
 import discord.utils
 import requests
 from discord import RequestsWebhookAdapter
 from discord import Webhook
 
 from rcon.game_logs import on_chat
+from rcon.game_logs import on_kill
+from rcon.game_logs import on_tk
 
 DISCORD_CHAT_WEBHOOK_URL = os.getenv("DISCORD_CHAT_WEBHOOK")
+DISCORD_KILLS_WEBHOOK_URL = os.getenv("DISCORD_KILLS_WEBHOOK")
 ALLOW_MENTIONS = os.getenv("DISCORD_CHAT_WEBHOOK_ALLOW_MENTIONS")
+PING_TRIGGER_WORDS = os.getenv("DISCORD_PING_TRIGGER_WORDS")
+PING_TRIGGER_ROLES = os.getenv("DISCORD_PING_TRIGGER_ROLES")
+SEND_KILLS = os.getenv("DISCORD_SEND_KILL_UPDATES")
+SEND_TEAM_KILLS = os.getenv("DISCORD_SEND_TEAM_KILL_UPDATES")
 
-logger = logging.getLogger(__name__)
+STEAM_PROFILE_URL = "http://steamcommunity.com/profiles/{id64}"
 
+RED = 0xA62019
+LIGHT_RED = 0xF93A2F
+BLUE = 0x006798
+LIGHT_BLUE = 0x0099E1
+GREEN = 0x07DA63
 
-@on_chat
-def post_chat_message_to_discord(_, log):
-    webhook_id, webhook_token = parse_webhook_url(DISCORD_CHAT_WEBHOOK_URL)
-    if not all([webhook_id, webhook_token]):
-        return
+KILL_ACTION_TO_COLOR = {
+    "KILL": GREEN,
+    "TEAM KILL": LIGHT_RED,
+}
+CHAT_ACTION_TO_COLOR = {
+    "[Axis][Team]": LIGHT_RED,
+    "[Axis][Unit]": RED,
+    "[Allies][Team]": LIGHT_BLUE,
+    "[Allies][Unit]": BLUE,
+}
 
-    try:
-        message = f"{log['action']}: {log['message']}"
-        if ALLOW_MENTIONS != 'yes':
-            message = discord.utils.escape_mentions(message)
-        message = discord.utils.escape_markdown(message)
+logger = logging.getLogger(__name__)
 
-        webhook = Webhook.partial(
-            id=webhook_id, token=webhook_token, adapter=RequestsWebhookAdapter()
-        )
-        logger.info("sending chat message len=%s to Discord", len(message))
-        action = log['action'].ljust(20)
-        player = log['player'].ljust(25)
-        side = '+' if 'Allies' in action else '-'
-        
-        webhook.send(f"```diff\n{side} {action} {player} [{log['steam_id_64_1']}]:\n{log['sub_content']}```")
 
-    except Exception as e:
-        logger.exception("error executing chat message webhook: %s", e)
+def escape_string(string):
+    string = discord.utils.escape_markdown(string)
+    string = discord.utils.escape_mentions(string)
+    return string
 
 
-@lru_cache
 def parse_webhook_url(url):
     """Parse and check validity of Discord webhook URL
     by performing a get request and checking existence
     of 'id' and 'token' in the JSON response.
     """
     if not url:
         logger.info("no Discord chat webhook URL provided")
-        return None, None
-
-    try:
-        resp = requests.get(url).json()
-        _id = int(resp["id"])
-        token = resp["token"]
-        return _id, token
-    except Exception as e:
-        logger.error("error parsing Discord chat webhook url: %s", e)
-        return None, None
+        return None
+    resp = requests.get(url).json()
```

Comment:
Caching is not totally removed, just moved, because `parse_webhook_url` is only ever called in `DiscordWebhookHandler.init_env_vars` and the value is stored in the class itself

Reply:
Someone else might however import `parse_webhook_url` and use it so I'll restore the LRU caching.